session.go

```diff
@@ -31,3 +31,30 @@ type Session interface {
 	// session implementation.
 	Version() (msize int, version string)
 }
+
+// WriteFull writes the whole p slice to the specified fid
+func WriteFull(s Session, ctx context.Context, fid Fid, p []byte, offset int64) error {
+	for len(p) > 0 {
+		written, err := s.Write(ctx, fid, p, offset)
+		if err != nil {
+			return err
+		}
+		p = p[written:]
+		offset += int64(written)
+	}
+	return nil
+}
+
+// ReadFull reads the whole file and returns it in a slice
+func ReadFull(s Session, ctx context.Context, fid Fid, offset int64) (p []byte, err error) {
+	var buf [4096]byte
```

Comment:
Right here is the problem: this fixes the buffer size and removes control from the caller.